internal/config/config.go

```diff
@@ -45,10 +40,31 @@ type Crypt4gh struct {
 	Passphrase string
 }
 
-// NewConfig initializes and parses the config file and/or environment using
+// New initializes and parses the config file and/or environment using
 // the viper library.
-func NewConfig() *Config {
-	parseConfig()
+func New(app string) *Config {
+	viper.SetConfigName("config")
+	viper.AddConfigPath(".")
+	viper.AutomaticEnv()
+	viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+	viper.SetConfigType("yaml")
+	if viper.IsSet("configPath") {
+		cp := viper.GetString("configPath")
+		ss := strings.Split(strings.TrimLeft(cp, "/"), "/")
+		viper.AddConfigPath(path.Join(ss...))
+	}
+
+	if viper.IsSet("configFile") {
+		viper.SetConfigFile(viper.GetString("configFile"))
+	}
+
+	if err := viper.ReadInConfig(); err != nil {
+		if _, ok := err.(viper.ConfigFileNotFoundError); ok {
+			log.Infoln("No config file found, using ENVs only")
```

Comment:
so it is either file or env only, no more combinations. that is ok. just wanted to double check.

Reply:
It's and/or, env overrules file values.